driver/src/main/com/mongodb/client/MongoCollection.java

```diff
@@ -61,7 +65,40 @@
      *
      * @return the collection options
      */
-    MongoCollectionOptions getOptions();
+    OperationOptions getOptions();
+
+    /**
+     * Create a new MongoCollection instance with the updated read preference.
+     *
+     * @param readPreference the new {@link com.mongodb.ReadPreference} for the collection
+     * @return a new MongoCollection instance with the updated readPreference
+     */
+    MongoCollection<T> withReadPreference(ReadPreference readPreference);
+
+    /**
+     * Create a new MongoCollection instance with the updated write concern.
+     *
+     * @param writeConcern the new {@link com.mongodb.WriteConcern} for the collection
+     * @return a new MongoCollection instance with the updated writeConcern
+     */
+    MongoCollection<T> withWriteConcern(WriteConcern writeConcern);
+
+    /**
+     * Create a new MongoCollection instance with the updated codec registry.
+     *
+     * @param codecRegistry the new {@link org.bson.codecs.configuration.CodecRegistry} for the collection
+     * @return a new MongoCollection instance with the updated codec registry
+     */
+    MongoCollection<T> withCodecRegistry(CodecRegistry codecRegistry);
+
+    /**
+     * Create a new MongoCollection instance with the updated default class to cast any documents returned from the database into..
+     *
+     * @param clazz the default class to cast any documents returned from the database into.
+     * @param <C> The type that the new collection will encode documents from and decode documents to
+     * @return a new MongoCollection instance with the updated default class
+     */
+    <C> MongoCollection<C> withClazz(Class<C> clazz);
```

Comment:
withClass is fine.  clazz as a variable nam is only used because class is a keyword.